src/ddx_lpb.f90

```diff
@@ -48,27 +49,22 @@ subroutine ddlpb(params, constants, workspace, state, phi_cav, e_cav, &
     real(dp), intent(out) :: esolv
     real(dp), intent(out), optional :: force(3, params % nsph)
     real(dp), intent(in), optional :: hessianphi_cav(3, 3, constants % ncav)
+    type(ddx_error_type), intent(inout) :: error
 
     call ddlpb_setup(params, constants, workspace, state, phi_cav, &
-        & e_cav, psi)
-    if (workspace % error_flag .eq. 1) return
-    call ddlpb_guess(params, constants, workspace, state, tol)
-    if (workspace % error_flag .eq. 1) return
-    call ddlpb_solve(params, constants, workspace, state, tol)
-    if (workspace % error_flag .eq. 1) return
+        & e_cav, psi, error)
+    call ddlpb_guess(params, constants, workspace, state, tol, error)
+    call ddlpb_solve(params, constants, workspace, state, tol, error)
```

Comment:
Is this a good idea? What happens if the guess errors? Will the solve just plain return?